providers/job.rb

```diff
@@ -116,17 +116,70 @@ def tweak_existing(job)
   # we need to specify uuid, not id to be able to update
   job.delete('id')
 
-  # rundeck gem returns project in context instead of directly project
-  # which is required to create job
-  context = job.delete('context')
-  job['project'] = context['project'] if context
+  #
+  # HACKS
+  #
+  # What follows is a sequence of hacks which should all be removed once the
+  # rundeck-ruby gem implements getting jobs in YAML, and not only in XML (as
+  # RunDeck outputs a different document hierarchy depending on the format).
+  #
+
+  # threadcount is returned as a string instead of an integer
+  unless job['dispatch'].nil? || job['dispatch']['threadcount'].nil?
+    if job['dispatch']['threadcount'].is_a?(String)
+      job['dispatch']['threadcount'] = job['dispatch']['threadcount'].to_i
+    end
+  end
+
+  # XML -> YAML translation (rundeck gem supports getting only xml)
+
+  # context field does not exist in YAML, but is present in XML
+  job.merge!(job.delete('context')) unless job['context'].nil?
```

Comment:
* Yes, as the context has several fields which would otherwise be left out.
* fair enough